crypto/x509/x_crl.c

```diff
@@ -103,13 +103,17 @@ static int crl_set_issuers(X509_CRL *crl)
 
         if (gtmp) {
             gens = gtmp;
```

Comment:
I would move this to the end of the `if()` statement, since it's only used if all the checks pass. Having it at the beginning implies that `gens` may be used inside the `if()`. Possibly even with a `gtmp = NULL` statement to make it clear that `gtmp` is no longer used, and does not need to be freed.

Reply:
can we do this in a follow-up re-factoring patch?